python/src/iceberg/expression/literals.py

```
@@ -0,0 +1,601 @@
+
+    def __str__(self):
+        return f'"{self.value}"'
+
+
+class UUIDLiteral(ComparableLiteral):
+    def __init__(self, value):
+        super(UUIDLiteral, self).__init__(f"UUIDLiteral({value})", value)
+
+    @singledispatchmethod
+    def to(self, type_var):
+        return None
+
+    @to.register(UUIDType)
+    def _(self, type_var):
+        return self
+
+
+class FixedLiteral(BaseLiteral):
+    def __init__(self, value):
+        super(FixedLiteral, self).__init__(f"FixedLiteral({value})", value)
+
+    @singledispatchmethod
+    def to(self, type_var):
+        return None
+
+    @to.register(FixedType)
+    def _(self, type_var):
+        if len(self.value) == type_var.length:
+            return self
+        return None
+
+    @to.register(BinaryType)
+    def _(self, type_var):
+        return BinaryLiteral(self.value)
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if other is None:
+            return False
+
+        return self.value < other.value
+
+    def __gt__(self, other):
+        if other is None:
+            return True
+
+        return self.value > other.value
+
+    def __le__(self, other):
+        if other is None:
+            return False
+
+        return self.value <= other.value
+
+    def __ge__(self, other):
+        if other is None:
+            return True
+
+        return self.value >= other.value
+
+
+class BinaryLiteral(BaseLiteral):
+    def __init__(self, value):
+        super(BinaryLiteral, self).__init__(f"BinaryLiteral({value})", value)
+
+    @singledispatchmethod
+    def to(self, type_var):
+        return None
+
+    @to.register(BinaryType)
+    def _(self, type_var):
+        return self
+
+    @to.register(FixedType)
+    def _(self, type_var):
+        if type_var.length == len(self.value):
+            return FixedLiteral(self.value)
+        return None
```

This should throw an exception if conversion fails.